File: nim/src/lightpkg/material.nim
import color

type Material* = tuple[refraction: float32, reflection: float32, color: Color]
  
const Black* = (refraction: 1.0'f32, reflection: 0'f32, color: color.Black)
const Red* = (refraction: 1.0'f32, reflection: 0'f32, color: color.Red)
const White* = (refraction: 1.0'f32, reflection: 0'f32, color: color.White)
const Green* = (refraction: 1.0'f32, reflection: 0'f32, color: color.Green)
const Blue* = (refraction: 1.0'f32, reflection: 0'f32, color: color.Blue)
const Yellow* = (refraction: 1.0'f32, reflection: 0'f32, color: color.Yellow)
const Cyan* = (refraction: 1.0'f32, reflection: 0'f32, color: color.Cyan)
const Magenta* = (refraction: 1.0'f32, reflection: 0'f32, color: color.Magenta)
const Glass* = (refraction: 1.0'f32, reflection: 0'f32, color: color.White)

File: light/src/light/material.rs
use crate::light::color::*;

#[derive(Debug)]
pub enum Material {
    Simple(Color),
    Reflective(Color, f32),
}

impl Material {
    pub fn red() -> Material {
        Material::Simple(Color(1.0, 0.0, 0.0))
    }
    pub fn green() -> Material {
        Material::Simple(Color(0.0, 1.0, 0.0))
    }
    pub fn blue() -> Material {
        Material::Simple(Color(0.0, 0.0, 1.0))
    }
    pub fn yellow() -> Material {
        Material::Simple(Color(1.0, 1.0, 0.0))
    }
    pub fn magenta() -> Material {
        Material::Simple(Color(1.0, 0.0, 1.0))
    }
    pub fn cyan() -> Material {
        Material::Simple(Color(0.0, 1.0, 1.0))
    }
    pub fn white() -> Material {
        Material::Simple(Color(1.0, 1.0, 1.0))
    }
}
